server/src/com/cloud/network/ExternalDeviceUsageManagerImpl.java

```diff
@@ -342,6 +342,12 @@ public ExternalDeviceNetworkUsageTask() {
 
         @Override
         protected void runInContext() {
+            //Check if there are any external deivces
+            //Skip external device usage collection if none exist
```

Comment:
@kishankavala Could you please change from "deivces" to "devices"?

Also, you can use this commented lines as a Javadoc block describing the runInContext() method (if you think it would improve your code).

Except that typo, your code seems ok.